pkg/debug/checkups/quarantine.go

```diff
@@ -0,0 +1,191 @@
+package checkups
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+type quarantine struct {
+	status           Status
+	summary          string
+	quarantineCounts map[string]int
+	dirsChecked      int
+}
+
+func (q *quarantine) Name() string {
+	return "Quarantine"
+}
+
+func (q *quarantine) Run(ctx context.Context, extraFh io.Writer) error {
+	q.quarantineCounts = make(map[string]int)
+
+	var (
+		quarantinePathDepths = map[string]int{
+			`C:\Windows\System32\Drivers`: 3,
+			`C:\ProgramData`:              3,
+
+			`/Library/Application Support`: 4,
+		}
+
+		meddlesomeProcessPatterns = []string{
+			`crowdstrike`,
+			`opswat`,
+			`defend`,
+			`defense`,
+			`threat`,
+			`virus`,
+			`quarantine`,
+			`snitch`,
+		}
+	)
+
+	fmt.Fprint(extraFh, "starting quarantine check\n")
+	q.logMeddlesomeProccesses(ctx, extraFh, meddlesomeProcessPatterns)
+	fmt.Fprintf(extraFh, "\nsearching for quarantined files:\n")
+
+	for path, maxDepth := range quarantinePathDepths {
+		fileInfo, err := os.Stat(path)
+		if err != nil {
+			fmt.Fprintf(extraFh, "%s does not exist\n", path)
+			continue
+		}
+
+		if !fileInfo.IsDir() {
+			fmt.Fprintf(extraFh, "expected %s to be a directory, but was not\n", path)
+			continue
+		}
+
+		q.checkDirs(extraFh, 0, maxDepth, path, "quarantine")
+	}
+
+	fmt.Fprintf(extraFh, "total directories checked: %d\n", q.dirsChecked)
+
+	if len(q.quarantineCounts) == 0 {
+		q.status = Passing
+		q.summary = "no quarantine directories found"
+		fmt.Fprint(extraFh, "no quarantine directories found\n")
+		return nil
+	}
+
+	fmt.Fprintf(extraFh, "quarantine directory paths and file counts:\n")
+
+	totalQuarantinedFiles := 0
+
+	for path, count := range q.quarantineCounts {
+		fmt.Fprintf(extraFh, "%s: %d\n", path, count)
+		totalQuarantinedFiles += count
+	}
+
+	if totalQuarantinedFiles == 0 {
+		q.status = Passing
+		q.summary = "no files found in quarantine directories"
+		return nil
+	}
+
+	q.status = Failing
```

Comment:
What about looking specifically for launcher/launcher.exe and osqueryd/osqueryd.exe being quarantined, and only setting the status to `Failing` if one of those is quarantined? I don't know if some other non-Kolide-related file being quarantined should result in a `Failing` status.

Reply:
Same as above, AVs renamed the files to GUIDs and store metadata some where so can't search for osquery directly. I actually started off on this path. However, it raises and interesting question. Should we ever fail this check or should we keep at as information purposes only?

Reply:
```suggestion
	q.status = Warning
```

I'm not sure we know enough to declare this as a failing. It's a warning or informational

Reply:
updated to warning